### DISCUSSION TITLE
Platform identification

### DISCUSSION THREAD

**Discussion by IOTSecurityHQ** [gramineproject/gramine#1298] [Q&A]
Does gramine support identification of platform enclave is running ?

I think UID value that serve that purpose. Not sure if gramine provide an interface to retrieve it.
 Function to read the UID value from the SGX enclave
void read_uid(uint64_t* uid) {
    uint32_t eax = 0x12;
    uint32_t ecx = 0;
    uint32_t ebx, edx;

    __asm__ __volatile__(
        "cpuid"
        : "=b"(ebx), "=d"(edx)
        : "a"(eax), "c"(ecx)
    );

    *uid = (((uint64_t)edx) << 32) | ebx;

**Comment by mkow** [comment]
No, this is intentionally impossible on SGX - SGX by design tries to make hardware fingerprinting hard. You can call `cpuid`, but it will be the untrusted host giving you the results.

**Reply by kailun-qin** [reply]
Yep, citing from Intel SGX Developer Guide - "The CPUID instruction is also illegal inside the enclave. Thus software that retrieves CPUID information must do so outside the enclave. Therefore, this information cannot be assured from a security viewpoint and should be used carefully".

Pls kindly find more info here: https://download.01.org/intel-sgx/latest/linux-latest/docs/Intel_SGX_Developer_Guide.pdf by searching w/ keyword `CPUID information`.
